docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,162 @@
+Self-nomination are fine.
+Decision is made by a consensus in the mailing list or via voting at the link:https://jenkins.io/project/governance-meeting/[governance meeting].
+* **Core Maintainers** - same process as for Core PR reviewers.
+  All nominees must also sign an link:https://github.com/jenkinsci/infra-cla/[Individual Contributor License Agreement] before getting the permission in GitHub repositories.
+* **Release Team Members** are assigned by the Jenkins Release officer
+
+== Pull request review process
+
+Jenkins core is a mission-critical part of the ecosystem.
+We need to ensure that submitted pull requests are not only code complete,
+but also that they do not introduce undesired defects, breaking changes and technical debt.
+At the same time, we are interested to make the review process as simple as possible for contributors and maintainers.
+
+=== Code review goals
+
+* Ensure that the pull request is feasible.
+** Not every change needs to be merged into the core, many features would be rather introduced as plugins.
+   Separate release cycles for plugins also help to deliver changes faster.
+** We want to extend the Jenkins core and incorporate widely used functionality there,
+   but we try to keep the core as minimal as possible in terms of dependencies.
+* Ensure that there is a consensus about the suggested change.
+** Not all changes are discussed before they submitted as pull requests.
+   Developer mailing lists, Jira issues and JEPs are used for discussions,
+   but sometimes the changes go straight to the pull requests.
+   And we are fine with that, especially for small patches.
+** Pull requests often become a venue to discuss feasibility, underlying technical decisions and design.
+   We are fine with that as well
+** If there is no consensus about the feasibility and implementation,
+   code reviewers are expected to suggest proper channels to maintainers
+* Ensure that the pull request matches the code quality practices: test automation, Javadoc for newly introduced APIs, etc.
+** There is no goal to make the code perfect tho
+* Ensure that the pull requests contain user-friendly changelogs and, if needed, upgrade guidelines
+* Last but not least,
+  guide newcomer contributors who are not familiar with the project processes.
+
+=== Code review non-goals
+
+* Enforce a particular coding style
+** Jenkins core has a complex codebase created by many contributors,
+   and different files have different designs
+** Main goal is 
+* Make contributors to fix unrelated issues
+** Create follow-up issues instead, it is fine to referene them in comments so that the contributor might want to pick them up
+* Breakdown the commit history or squashing PRs
+** Not every contributor is a Git expert, do not request changes in the commit history unless it is necessary
+** Core maintainers can squash PRs during the merge.
+   If you feel this is important, use the `squash-me-please` label
```

Comment:
When would it be important though?

Needs some guideline. Something like "An indicator to use merge commits would be if the PR history cleanly separates steps in developing the complete solution, that are individually useful to retain for the future. Substantial iteration after code review feedback, or a generally 'unclean' history (e.g. commits that fix compile errors, have no useful messages, or only exist to rebuild the PR) would indicate a preference for squash-merging the PR" perhaps?